### ISSUE TITLE
Ability to toggle between "markers" and "shaded regions" to display locations

### ISSUE THREAD

**Issue by jywarren** [publiclab/leaflet-blurred-location-display#10]
When showing "shaded regions" clicking in a shaded region could pop up a list of the points within that region? 

Perhaps as a break-out, shaded regions could be darker if there are more points within?

**Comment by mridulnagpal** [comment]
@jywarren Like a pop-up right? For getting list of all points in that region.

**Comment by mridulnagpal** [comment]
@publiclab/soc @jywarren How do I get all points between 4 lat,lng points using the API, or till what radius are the points fetched from the center lat,lng?

**Comment by jywarren** [comment]
Maybe this is a question for @milaaraujo @stefannibrasil ? I think we can search for people and nodes in a bounding box in the API... or maybe we need to add this?

**Comment by stefannibrasil** [comment]
hi, everyone!

we would need to add that, right now it is only getting the nodes near the given coordinates (lat and lon points):

https://github.com/publiclab/plots2/blob/5574837fa2ada7f253d401d65469fa7ec39566fb/app/services/search_service.rb#L130-L166

I am not really sure how the radius is calculated there :thinking:

**Comment by stefannibrasil** [comment]
https://github.com/publiclab/plots2/blob/5574837fa2ada7f253d401d65469fa7ec39566fb/app/services/search_service.rb#L130-L166

**Comment by stefannibrasil** [comment]
https://github.com/publiclab/plots2/blob/5574837fa2ada7f253d401d65469fa7ec39566fb/app/services/search_service.rb#L130-L136

**Comment by stefannibrasil** [comment]
don't know why the code snippet isn't working, but there is the definition for the endpoint that get the nearby nodes given a pair of coordinates

**Comment by stefannibrasil** [comment]
I am not sure why the code snippet isn't working, but there you can see that it only show the nearby nodes given a pair of coordinates... all of this to say that we can work on that in the next weeks!

**Comment by mridulnagpal** [comment]
@stefannibrasil @jywarren Sounds good as soon as the radius is known we can work on getting all the nodes inside a box.

**Comment by jywarren** [comment]
OK, great! I also made a gh-pages at https://publiclab.github.io/leaflet-blurred-location-display/examples/index.html although it may take a little while to show up

**Comment by jywarren** [comment]
There we go!

**Comment by jywarren** [comment]
https://publiclab.github.io/leaflet-blurred-location-display/examples/index.html

**Comment by mridulnagpal** [comment]
@jywarren Is the API done?

**Comment by jywarren** [comment]
No, but just as you've added example locations, you can add ones that
specify precision, and develop this display around the example data, don't
you think?

**Comment by mridulnagpal** [comment]
@jywarren Got it, working on it!

**Comment by jywarren** [comment]
Breaking out an issue to develop ways to show multiple data points that have been blurred as a kind of "heat map" or chloropleth?

https://deutchland.maps.luftdaten.info/#4/35.03/5.40 does this in Leaflet, although with a hex grid...

![screen shot 2018-11-27 at 3 45 15 pm](https://user-images.githubusercontent.com/24359/49110467-9759f180-f25b-11e8-8fb8-66db589ea972.png)

There are for sure Leaflet chloropleth map plugins out there!

**Comment by sagarpreet-chadha** [comment]
This is really cool 😄 . Will start work on this tomorrow 👍 !
